hetmech/path_count.py

```diff
@@ -0,0 +1,135 @@
+import collections
+from scipy.sparse import csr_matrix
+import scipy.sparse
+
+from .matrix import metaedge_to_adjacency_matrix
+from .degree_weight import dwwc_step
+
+
+def dwpc(graph, metapath, damping=1.0, verbose=False):
+    """
+    Compute the degree-weighted path count (DWPC) on specified metapath
+    and graph, with the paths normalized by the given damping parameter.
+
+    This function computes the dwpc for all paths of the given metpath
+    type, via matrix multiplies and corrections to handle non-path walks.
+    
+    NOTE: this function can handle metapaths that do not repeat
+    metatnode typs, and metapaths that repeat exactly one node type
+    any number of times, but cannot handle metapaths that repeat
+    more than one type of node.
+    
+    Parameters
+    ==========
+    graph : hetio.hetnet.Graph
+        graph to extract adjacency matrices along
+    metapath : hetio.hetnet.MetaPath
+        metapath for which function computes DWPCs
+    damping : scalar
+        exponent of degree in degree-weighting of DWPC
+    verbose : bool
+        set to True to have function print to screen
+        (temporary, for debugging)
+    """
+
+    # Check sequence of metanode types for repeats
+    nodetype_sequence = collections.defaultdict(int)
+    for metaedge in metapath:
+        nodetype_sequence[metaedge.source] += 1
+        nodetype_sequence[metaedge.target] += 1
```

Comment:
When I originally explored this, after your initial comment, I couldn't get things working --- I just tried again and figured out I wasn't converting the output of `metapath.get_nodes()` to strings properly. Now it's working, and the code looks cleaner for it :-)